web-extensions/chrome/scripts/load-bookmarks.js

```diff
@@ -1,49 +1,116 @@
-function generateBookmarkList(bookmarks) {
+const uriParams = document.location.search.split('?')[1].split('&');
+const comicName = decodeURI(uriParams[0].split('=')[1]);
+const chapter = uriParams[1].split('=')[1];
+const page = uriParams[2].split('=')[1];
+console.log(`BmcSideBar: comic ${comicName}, chapter=${chapter}, page=${page}`);
+
+function BmcMangaList() {
+    this._node = document.getElementById('manga-list');
+}
+
+BmcMangaList.prototype.onEntryClick = function(comicId) {
+    console.log('clickedOnManga!');
+    window.top.postMessage({comicId}, '*');
+}
+
+BmcMangaList.prototype.generate = function() {
+    var bookmarks = [
+        {label: "naruto",           id: 0},
+        {label: "bleach",           id: 1},
+        {label: "one piece",        id: 2},
+        {label: "goblin slayer",    id: 3},
+        {label: "hunter x hunter",  id: 4},
+    ];
+
     console.log("generating bookmark list");
     var mangaList = document.getElementById("manga-list");
     for (var i = 0; i < bookmarks.length; ++i) {
         var manga = document.createElement('div');
-        manga.innerText = bookmarks[i];
-        manga.onclick = function() {
-            clickedOnManga(this.innerText);
-        };
+        manga.innerText = bookmarks[i].label;
+        manga.onclick = () => this.onEntryClick(bookmarks[i].id);
         mangaList.appendChild(manga);
     }
 }
 
-function clickedOnManga(comicName) {
-    const engine = new BmcEngine();
-    engine.addEventListener(engine.events.register.error, err => {
-    });
-    engine.addEventListener(engine.events.register.complete, () => {
-        console.log('clickedOnManga!');
-        window.top.postMessage(`{"comicName": "${comicName.split('"').join('\\"')}"}`, '*');
-    });
-    engine.register(comicName, null, null);
+BmcMangaList.prototype.hideEntry = function(entry) {
+    entry.style.display = 'none';
 }
 
-function addEvents() {
-    var but = document.getElementById('hide-but');
-    but.onclick = function() {
-        showHideSidePanel(but);
+BmcMangaList.prototype.showEntry = function(entry) {
+    entry.style.display = '';
+}
+
+BmcMangaList.prototype.match = function(value, match) {
+    var lvalue = value.toLowerCase();
+    var lmatch = match.toLowerCase();
+    for (var i=0; i < lmatch.length; ++i) {
```

Comment:
`i=0` isn't very beautiful. ;)

Reply:
But `childNodes` is neither proper english nor a proper array